digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala

```diff
@@ -203,6 +203,15 @@ case class ProjectLink(id: Long, roadNumber: Long, roadPartNumber: Long, track:
       case (Some(cp1),Some(cp2)) => cp1.source
     }
   }
+
+  def hasCalibrationPointAt(addressMValue: Long): Boolean = {
+    calibrationPoints match {
+      case (None, None) => false
+      case (Some(cp1), None) => cp1.addressMValue == addressMValue
+      case (None, Some(cp1)) => cp1.addressMValue == addressMValue
+      case (Some(cp1), Some(cp2)) => cp1.addressMValue == addressMValue || cp2.addressMValue == addressMValue
+    }
```

Comment:
You can change the code to be like that:
```
calibrationPoints match {
      case (None, None) => false
      case (Some(cp1), _) => cp1.addressMValue == addressMValue
      case (_, Some(cp1)) => cp1.addressMValue == addressMValue
```

Reply:
Not in this case because the value we are trying to find can be in either of the calibration points, if we replace by that, the last expression becomes unreachable and we become unable to find the value we seek when we have calibration points in both ends.